server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2188,7 +2190,10 @@ public void onFailure(Exception e) {
 
                     @Override
                     protected void doRun() {
-                        flush(new FlushRequest().waitIfOngoing(false).force(false));
+                        if (flush(new FlushRequest().waitIfOngoing(false).force(false)) == false) {
+                            // In case the flush did not happen, revert active flag so that a next flushOnIdle request can happen (#87888)
```

Comment:
```suggestion
                            // In case an ongoing flush was detected, revert active flag so that a next flushOnIdle request will retry (#87888)
```